### PR TITLE
VTCCA-3130 matching modernised contract for the organisation

### PR DESCRIPTION
**Pull request by DanielConnelly**

### CODE REVIEW

**Comment by darrell-plant** [comment]
LGTM 👍

File: app/controllers/agentAppointment/AppointAgentController.scala [hunk 1]
```diff
@@ -255,9 +256,9 @@ class AppointAgentController @Inject()(
                 .map(oar => oar.copy(filterTotal = oar.authorisations.size))
                 .map(oar => oar.copy(authorisations = oar.authorisations.take(pagination.pageSize)))
             } yield {
-              BadRequest(views.html.propertyrepresentation.revokeAgentProperties(Some(errors), AppointAgentPropertiesVM(group, response), PaginationParameters(), GetPropertyLinksParameters(), group.agentCode))
+              BadRequest(views.html.propertyrepresentation.revokeAgentProperties(Some(errors), AppointAgentPropertiesVM(group, response), PaginationParameters(), GetPropertyLinksParameters(), agentCode))
             }
-          case None =>
+          case None | Some(_) =>
```
Comment (PeterPerhac) [inline app/controllers/agentAppointment/AppointAgentController.scala]: ` case None | Some(_) ` ?  if it either is or isn't there?  use `case _` for catch-all

File: app/controllers/agentAppointment/AppointAgentController.scala [hunk 2]
```diff
@@ -198,9 +199,9 @@ class AppointAgentController @Inject()(
       },
       success = (agent: AgentId) => {
         accounts.withAgentCode(agent.id) flatMap {
-          case Some(group) =>
-            Future.successful(Redirect(routes.AppointAgentController.selectAgentPropertiesSearchSort(PaginationParameters(), GetPropertyLinksParameters(), group.agentCode)))
-          case None =>
+          case Some(AgentGroupAccount(_, agentCode)) =>
+            Future.successful(Redirect(routes.AppointAgentController.selectAgentPropertiesSearchSort(PaginationParameters(), GetPropertyLinksParameters(), agentCode)))
+          case None | Some(_) =>
```
Comment (habeeb01) [inline app/controllers/agentAppointment/AppointAgentController.scala]: case _ for catch all

File: app/models/GroupAccount.scala [hunk 3]
```diff
@@ -18,9 +18,25 @@ package models
...
+                         isAgent: Boolean,
+                         agentCode: Option[Long]
+                       )
 
 object GroupAccount {
   implicit val format = Json.format[GroupAccount]
+
+  object AgentGroupAccount {
+    def unapply(account: GroupAccount): Option[(GroupAccount, Long)] =
+      account
```
Comment (PeterPerhac) [inline app/models/GroupAccount.scala]: `PartialFunction.condOpt(account){ case a@GroupAccount(_, _, _, _, _, _, true, code) => (a, code)}` avoids mapping the agent code to a tuple if the account is not actually an agent account. but i guess it's less readable

**Comment by PeterPerhac** [comment]
some small nitpicks but otherwise looks good to me 👍

**Pull request merged**